Program codegen generates invalid TypeScript lambdas.

<!-- Please provide a general summary of the issue, including what you're trying to accomplish. -->
The `aws-eks-pp` TypeScript test fails with parsing errors. We emitted `zones.names.map((k, v) => {key: k, value: v})` which is invalid. We should have emitted `zones.names.map((k, v) => ({key: k, value: v}))`
## Steps to reproduce
<!-- Provide a link to a live example or an unambiguous set of steps to reproduce this bug. Tell us what you expected to happen, and what happened instead. Please include code to reproduce, if relevant. -->

1. `cd pkg/codegen/internal/test/testdata/aws-eks-pp`
2. `yarn run tsc --noEmit aws-eks.ts`

Expected: Clean compile
Actual: 
```sh
yarn run v1.22.11
warning package.json: No license field
$ /Users/ianwahbe/Projects/pulumi/pulumi/pkg/codegen/internal/test/testdata/aws-eks-pp/node_modules/.bin/tsc --noEmit aws-eks.ts
aws-eks.ts(34,65): error TS1005: ';' expected.
aws-eks.ts(47,65): error TS1005: ';' expected.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Note: there are probably other errors behind this test. We can file file issues for those when this test is solved.

--- a/pkg/codegen/nodejs/gen_program_expressions.go
+++ b/pkg/codegen/nodejs/gen_program_expressions.go
@@ -373,11 +373,12 @@ func (g *generator) GenFunctionCallExpression(w io.Writer, expr *model.FunctionC
 				g.genRange(w, call, true)
 				return
 			}
-			g.Fgenf(w, "%.20v.map((k, v)", expr.Args[0])
+			// Mapping over a list with a tuple receiver accepts (value, index).
+			g.Fgenf(w, "%.20v.map((v, k)", expr.Args[0])
 		case *model.MapType, *model.ObjectType:
 			g.Fgenf(w, "Object.entries(%.v).map(([k, v])", expr.Args[0])
 		}
-		g.Fgenf(w, " => {key: k, value: v})")
+		g.Fgenf(w, " => ({key: k, value: v}))")
 	case "fileArchive":
 		g.Fgenf(w, "new pulumi.asset.FileArchive(%.v)", expr.Args[0])
 	case "remoteArchive":

--- a/pkg/codegen/testing/test/program_driver.go
+++ b/pkg/codegen/testing/test/program_driver.go
@@ -77,8 +77,6 @@ var PulumiPulumiProgramTests = []ProgramTest{
 	{
 		Directory:   "aws-eks",
 		Description: "AWS EKS",
-		SkipCompile: codegen.NewStringSet("nodejs"),
-		// Blocked on nodejs: TODO[pulumi/pulumi#8067]
 	},
 	{
 		Directory:   "aws-fargate",

--- a/pkg/codegen/testing/test/testdata/aws-eks-pp/nodejs/aws-eks.ts
+++ b/pkg/codegen/testing/test/testdata/aws-eks-pp/nodejs/aws-eks.ts
@@ -31,7 +31,7 @@ export = async () => {
     // Subnets, one for each AZ in a region
     const zones = await aws.getAvailabilityZones({});
     const vpcSubnet: aws.ec2.Subnet[] = [];
-    for (const range of zones.names.map((k, v) => {key: k, value: v})) {
+    for (const range of zones.names.map((v, k) => ({key: k, value: v}))) {
         vpcSubnet.push(new aws.ec2.Subnet(`vpcSubnet-${range.key}`, {
             assignIpv6AddressOnCreation: false,
             vpcId: eksVpc.id,
@@ -44,7 +44,7 @@ export = async () => {
         }));
     }
     const rta: aws.ec2.RouteTableAssociation[] = [];
-    for (const range of zones.names.map((k, v) => {key: k, value: v})) {
+    for (const range of zones.names.map((v, k) => ({key: k, value: v}))) {
         rta.push(new aws.ec2.RouteTableAssociation(`rta-${range.key}`, {
             routeTableId: eksRouteTable.id,
             subnetId: vpcSubnet[range.key].id,

--- a/pkg/codegen/testing/test/testdata/aws-s3-folder-pp/nodejs/aws-s3-folder.ts
+++ b/pkg/codegen/testing/test/testdata/aws-s3-folder-pp/nodejs/aws-s3-folder.ts
@@ -9,7 +9,7 @@ const siteBucket = new aws.s3.Bucket("siteBucket", {website: {
 const siteDir = "www";
 // For each file in the directory, create an S3 object stored in `siteBucket`
 const files: aws.s3.BucketObject[] = [];
-for (const range of fs.readDirSync(siteDir).map((k, v) => {key: k, value: v})) {
+for (const range of fs.readDirSync(siteDir).map((v, k) => ({key: k, value: v}))) {
     files.push(new aws.s3.BucketObject(`files-${range.key}`, {
         bucket: siteBucket.id,
         key: range.value,

